### DISCUSSION TITLE
How to extract all highlighted text from a pdf?

### DISCUSSION THREAD

**Discussion by TheoPallis** [pymupdf/PyMuPDF#1573] [Looking for help]
I need to extract all highlighted text from a pdf.

I have tried the following but it returns only an empty box.
```

import fitz
doc = fitz.open(r'C:\Users\Θοδωρής\newsql.pdf')

page = doc.load_page

for page in doc:
     for annot in page.annots():
        highlights = annot.get_text("text")
        print(highlights)
        
    

```

**Comment by JorjMcKie** [comment]
It **_cannot_** work like this!
The highlighted text is not part of the annotation, but part of the PDF itself, so you have to extract the text located inside the highlight rectangle: `text = page.get_textbox(annot.rect)`.

**Reply by TheoPallis** [reply]
Thank you, I will look into it and get back to you soon with any questions.
